src/tests/test_viewsGenerationSystem.py

```diff
@@ -79,89 +120,70 @@ def tearDown(self):
-
+		write_nvdaAPIVersions(nvdaAPIVersionsJson("2021.1.0", backCompatTo="2021.1.0"))
+		write_addons(addonJson('foo/0.1.1.json', "stable", required="2021.1.0", tested="2021.1.0"))
 		self.runTransformation()
 
 	def test_throw_error_on_nonempty_output_folder(self):
 		"""Confirms using an existing output directory throws an error
 		"""
 		# Make the folder before transform
 		Path(DATA_DIR.OUTPUT.value).mkdir(parents=True, exist_ok=True)
-		
+		write_nvdaAPIVersions()
 		with self.assertRaises(subprocess.CalledProcessError) as transformError:
 			self.runTransformation()
+
 		doubleEscapedDir = DATA_DIR.OUTPUT.replace('\\', '\\\\')  # stderr escapes all the backslashes twice
 		self.assertIn(
 			"FileExistsError: [WinError 183] Cannot create a file when that file already exists: "
 			f"'{doubleEscapedDir}'",
+			# requires that runTransformation and child processes log errors to stderr
 			transformError.exception.stderr.decode("utf-8")
 		)
 
-	def _assertAddonDataWritten(self, expectedPathToAddon: str, expectedAddonVersion: MajorMinorPatch):
-		"""Confirms that an addon is written to a path and has an expected version."""
-		self.assertTrue(Path(expectedPathToAddon).exists())
-		with open(expectedPathToAddon, "r") as expectedAddon:
-			addonData = json.load(expectedAddon)
-		self.assertDictEqual(addonData["addonVersionNumber"], expectedAddonVersion._asdict())
+	def _assertAddonDataWritten(self, expectedPathToAddon: str, expectedAddonVersionStr: str):
+		"""Confirms that an addon is written to a path and the file contains an expected version."""
+		fullPathToAddon = os.path.join(DATA_DIR.OUTPUT, expectedPathToAddon)
+		self.assertTrue(Path(fullPathToAddon).exists())
+		with open(fullPathToAddon, "r") as expectedAddonFile:
+			addonData = json.load(expectedAddonFile)
+		addonVersion = MajorMinorPatch(**addonData["addonVersionNumber"])
+		self.assertEqual(expectedAddonVersionStr, str(addonVersion))
 
 	def test_output_file_structure_matches_expected(self):
-		"""Confirms that a successful transform of multiple addons is written as expected."""
-		# older version
-		addon = MockAddon()
-		addon.addonId = "multipleVersionsAddon"
-		addon.channel = "stable"
-		addon.addonVersion = MajorMinorPatch(2, 1)
-		addon.minNvdaAPIVersion = MajorMinorPatch(2020, 1)
-		addon.lastTestedVersion = MajorMinorPatch(2020, 2)
-		_TestDataGenerator.write_mock_addon_to_input_dir(addon)
-
-		# newer version
-		addon = deepcopy(addon)
-		addon.addonVersion = MajorMinorPatch(13, 0)
-		addon.minNvdaAPIVersion = MajorMinorPatch(2020, 2)
-		addon.lastTestedVersion = MajorMinorPatch(2021, 1)
-		_TestDataGenerator.write_mock_addon_to_input_dir(addon)
-
+		"""Confirms that a transform of multiple addon versions is written as expected."""
+		write_nvdaAPIVersions(
+			nvdaAPIVersionsJson("2020.1.0", backCompatTo="2020.1.0"),
+			nvdaAPIVersionsJson("2020.2.0", backCompatTo="2020.2.0"),
+			nvdaAPIVersionsJson("2021.1.0", backCompatTo="2021.1.0"),
+		)
```

Comment:
Well, we would have to add input to `_assertAddonDataWritten` for each addon, for each compatible NVDA API version. This would make the tests dynamic, if we add compatible NVDA API versions we would have to update the tests.

Reply:
I am using the static file now, as we discussed.